videostore/renamer.go

```diff
@@ -0,0 +1,148 @@
+package videostore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"go.viam.com/rdk/logging"
+)
+
+// renamer watches a directory and converts local timestamps to unix timestamps
+type renamer struct {
+	watchDir  string
+	outputDir string
+	logger    logging.Logger
+}
+
+func newRenamer(watchDir, outputDir string, logger logging.Logger) *renamer {
+	return &renamer{
+		watchDir:  watchDir,
+		outputDir: outputDir,
+		logger:    logger,
+	}
+}
+
+// processSegments periodically scans the directory for new files to process
+//
+// This function polls the directory regularly, looking for MP4 files
+// and processing them in order of creation.
+func (r *renamer) processSegments(ctx context.Context) error {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	r.logger.Debugf("Starting to scan directory: %s", r.watchDir)
+	for {
+		select {
+		case <-ctx.Done():
+			r.logger.Debug("Context done, stopping scanner and flushing remaining files")
+			if err := r.close(); err != nil {
+				return fmt.Errorf("error during close: %w", err)
+			}
+			return nil
+		case <-ticker.C:
+			if err := r.scanAndProcessFiles(); err != nil {
+				r.logger.Errorf("Error scanning directory: %v", err)
+			}
+		}
+	}
+}
+
+// scanAndProcessFiles scans the watch directory for MP4 files and processes them
+//
+// It excludes the most recent file to avoid processing it while it's still
+// being written by the segmenter.
+func (r *renamer) scanAndProcessFiles() error {
+	files, err := r.getSortedMPEGFiles()
+	if err != nil {
+		return err
+	}
+	if len(files) <= 1 {
+		r.logger.Debug("one or fewer active files found, skipping processing")
+		return nil
+	}
+
+	return r.processFiles(files[:len(files)-1]) // Exclude the most recent file
```

Comment:
1. process all the files
2. you need to check if the file parses (if you can get it's duration). If you can't skip it and log at debug level